dist/invenio-records-js.js

```diff
@@ -0,0 +1,703 @@
+      var formTemplatesBase = attrs.formTemplatesBase;
+
+      if (formTemplatesBase.substr(formTemplatesBase.length -1) !== '/') {
+        formTemplatesBase = formTemplatesBase + '/';
+      }
+
+      angular.forEach(formTemplates, function(value, key) {
+        schemaFormDecorators
+          .decorator()[key.replace('_', '-')]
+          .template = formTemplatesBase + value;
+      });
+    }
+
+    var getProp = function (obj, prop) {
+      return prop.split('.').reduce(function(data, item) {
+        return data[item];
+      }, obj);
+    };
+
+    function _errorOrEmpty(){
+      var defer = $q.defer();
+      defer.resolve({data: []});
+      return defer.promise;
+    }
+
+    function _suggestEngine(args, map) {
+      if (args.url !== undefined) {
+        return InvenioRecordsAPI.request(args)
+          .then(
+            function success(response) {
+              var data = getProp(response.data, map.resultSource);
+              angular.forEach(data, function(value, key) {
+                var item = {};
+                item[map.valueProperty] = getProp(value, map.valueSource || map.valueProperty);
+                item[map.nameProperty] = getProp(value, map.nameSource || map.nameProperty);
+                data[key] = item;
+              });
+              return {
+                data: data
+              };
+            },
+            _errorOrEmpty
+          );
+      }
+      return _errorOrEmpty();
+    }
+
+    function _urlParser(url, urlParameters, query){
+      if (urlParameters !== undefined) {
+        var urlArgs = {};
+        angular.forEach(urlParameters, function(value, key) {
+          try {
+            if (value === 'value'){
+              urlArgs[key] = query;
+            } else {
+              urlArgs[key] = scope.$eval(value) || value;
+            }
+          } catch(error) {
+            urlArgs[key] = value;
+          }
+        });
+        url = url + '?' + $httpParamSerializerJQLike(
+          angular.merge({}, urlArgs)
+        );
+      }
+      return url;
+    }
+
+    function autocompleteSuggest(options, query) {
+      var args = {};
+      if (query === '') {
+        if (scope.lastSuggestions[options.url]) {
+          var defer = $q.defer();
+          defer.resolve(scope.lastSuggestions[options.url]);
+          return defer.promise;
+        } else if (options.scope && typeof options.scope.insideModel === 'string') {
+          query = options.scope.insideModel;
+          query = scope.$eval(options.processQuery || 'query', {query: query});
+        } else if (Array.isArray(options.initial) && options.initial.length > 0) {
+          var titleMap = options.initial.map(function(element) {
+            return {
+              "name": "(" + element["source"] + ") " + element['value'],
+              "value": element
+            }
+          });
+          var defer = $q.defer();
+          defer.resolve({"data": titleMap});
+          return defer.promise;
+        }
```

Comment:
The lines from 222-232 are new here, right?

Reply:
Yes. The content of `dist/` is all autogenerated from the content of `src/`.